model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/updater/MapJpaLiquibaseUpdaterProvider.java

```diff
@@ -32,13 +32,29 @@
 import liquibase.changelog.RanChangeSet;
 import liquibase.exception.LiquibaseException;
 import org.jboss.logging.Logger;
-import org.keycloak.connections.jpa.updater.liquibase.ThreadLocalSessionContext;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.map.storage.ModelEntityUtil;
 import org.keycloak.models.map.storage.jpa.updater.MapJpaUpdaterProvider;
 
 public class MapJpaLiquibaseUpdaterProvider implements MapJpaUpdaterProvider {
 
+    public static class ThreadLocalSessionContext {
+
+        private static final ThreadLocal<KeycloakSession> currentSession = new ThreadLocal<KeycloakSession>();
+    
+        public static KeycloakSession getCurrentSession() {
+            return currentSession.get();
+        }
+    
+        public static void setCurrentSession(KeycloakSession session) {
+            currentSession.set(session);
+        }
+    
+        public static void removeCurrentSession() {
+            currentSession.remove();
+        }
+    }
+    
```

Comment:
I remember testing the removal of this `ThreadLocalSessionContext` from the updater provider a while back. As far as I can tell this was used in the legacy module to provide a way for custom extensions to access the session, but so far in the new store we haven't needed any of that. It doesn't really hurt having this but I think this code is currently useless.

Reply:
That is correct. I slightly prefer keeping it here but if you prefer it removed together with related code, I'm happy to